test/python/tf_unittest_runner.py

```diff
@@ -39,89 +42,147 @@ def main():
     optional.add_argument(
         '--run_test',
         help=
-        "Runs the testcase and returns the output. Eg:math_ops_test.math_ops_test.DivNoNanTest.testBasic"
+        "Runs the testcase and returns the output. Eg:math_ops_test.DivNoNanTest.testBasic"
     )
     parser._action_groups.append(optional)
     arguments = parser.parse_args()
 
-    all_dirs_to_path(arguments.tensorflow_path)
+    accepted_formats = [
+        "math_ops_test", "mat_ops_test.DivNoNanTest",
+        "math_ops_test.DivNoNanTest.testBasic", "math_ops_test.DivNoNanTest.*",
+        "math_ops_test.D*", "math_ops_test.*", "math_*_test", "math_*_*_test",
+        "math*_test"
+    ]
 
     if (arguments.list_tests):
-        list_tests(arguments.list_tests)
+        try:
+            module_list = regex_walk(arguments.tensorflow_path,
+                                     arguments.list_tests)
+            list_tests(module_list, arguments.list_tests)
+        except:
+            print(
+                "Enter a valid argument to LIST tests.\nList of accepted formats:"
+            )
```

Comment:
`LIST tests`-->`--list_tests`

Reply:
So there is a try-except here because?
which of the function calls `regex_walk` or `list_tests` is expected to fail?